lib/jxl/dec_external_image.cc

```diff
@@ -408,6 +429,9 @@ Status ConvertToExternal(const jxl::ImageBundle& ib, size_t bits_per_sample,
             if (want_alpha) {
               row_in[c++] = ib.HasAlpha() ? alpha->Row(y) : ones.Row(0);
             }
+            if (want_extra_channel) {
```

Comment:
I know this is correct, but it would make it more obvious what's going on if you do `if (want_extra_channel) { row_in[c++] = extra->Row(y); } else { ... set color and alpha ... }`. I think the fact that this logic is repeated 3 times merits a refactor of this code, for example you can split setting up which Planes need to be converted and separately the code that actually converts them, transposes them, etc.

Reply:
No longer an issue after https://github.com/libjxl/libjxl/pull/325